### ISSUE TITLE
Idyll crashes with JavaScript heap out of memory

### ISSUE THREAD

**Issue by suhr** [idyll-lang/idyll#736]
**Describe the bug**

```
Building Idyll project with output directory: build


<--- Last few GCs --->

[7130:0x4199620]    61034 ms: Scavenge (reduce) 4012.4 (4042.2) -> 4011.9 (4043.4) MB, 6.8 / 0.0 ms  (average mu = 0.196, current mu = 0.120) allocation failure 
[7130:0x4199620]    61076 ms: Scavenge (reduce) 4012.5 (4044.9) -> 4012.2 (4045.4) MB, 6.6 / 0.0 ms  (average mu = 0.196, current mu = 0.120) allocation failure 
[7130:0x4199620]    61121 ms: Scavenge (reduce) 4012.8 (4042.4) -> 4012.4 (4043.7) MB, 6.7 / 0.0 ms  (average mu = 0.196, current mu = 0.120) allocation failure 


<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x969558 node::Abort() [node]
 2: 0x890af0  [node]
 3: 0xb0e1d2 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xb0e48e v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xcdb475  [node]
 6: 0xcdb545  [node]
 7: 0xced017 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [node]
 8: 0xced84e v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 9: 0xcee57c v8::internal::Heap::CollectAllGarbage(int, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
10: 0xc92d68 v8::internal::StackGuard::HandleInterrupts() [node]
11: 0x104560c v8::internal::Runtime_StackGuard(int, unsigned long*, v8::internal::Isolate*) [node]
12: 0x13edb99  [node]
```

**To Reproduce**
1. Clone https://github.com/suhr/prog-math/tree/cat/cat
2. Run `idyll watch`

**Expected behavior**
The document has only 1921 lines, building it should not cause OOM on a machine with 16 GB RAM.

**Comment by mathisonian** [comment]
Thanks for reporting this @suhr. It seems likely to be a compiler issue, but I I'm trying to track down where the memory issue is coming from. I'll keep this issue open and update as I learn more.

**Comment by mathisonian** [comment]
Fixed in #769; this will be available in the next release to npm
